### PR TITLE
Add feature flag for importing passwords via Google Password Manager

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/608920331025315/1207446004465151/f 

### Description
Sets up new remote config handling for autofill subfeature: `canImportFromGooglePasswordManager`

This dictates whether we offer up the option to import passwords directly from Google Password Manager or not. 

Additional config:
| Key | Type | Description |
|--------|--------|--------|
| `launchUrl` | `String` | The initial URL to launch for Google Password Manager |
| `javascriptConfig` | `String` | The config to pass onto the JS layer handling GPM imports |

### Steps to test this PR
- QA optional; they'll be tested more easily in the branches above

**Optional**
- [x] Apply patch below
- [x] Fresh install, launch app and allow time for remote config to be downloaded
- [x] Open `Device Explorer` and check values look correct in `shared_prefs/com.duckduckgo.feature.toggle.autofill.xml`

### Patch
```
Index: privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt b/privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt
--- a/privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt	(revision Staged)
+++ b/privacy-config/privacy-config-api/src/main/java/com/duckduckgo/privacy/config/api/PrivacyFeatureName.kt	(date 1730729845774)
@@ -27,4 +27,5 @@
     TrackingParametersFeatureName("trackingParameters"),
 }
 
-const val PRIVACY_REMOTE_CONFIG_URL = "https://staticcdn.duckduckgo.com/trackerblocking/config/v4/android-config.json"
+// in use for ship review build #2 const val PRIVACY_REMOTE_CONFIG_URL = "https://jsonblob.com/api/1299412335805194240"
+const val PRIVACY_REMOTE_CONFIG_URL = "https://jsonblob.com/api/1301563859532636160"
```

### CODE REVIEW

**Comment by CDRussell** [comment]
* **#4602** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4602?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4601** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4601?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5096** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/5096?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CDRussell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/importing/gpm/feature/AutofillImportPasswordSettings.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+@ContributesBinding(AppScope::class)
+class AutofillImportPasswordConfigStoreImpl @Inject constructor(
+    private val autofillFeature: AutofillFeature,
+    private val dispatchers: DispatcherProvider,
+) : AutofillImportPasswordConfigStore {
+
+    override suspend fun getConfig(): AutofillImportPasswordSettings {
+        return withContext(dispatchers.io()) {
+            with(autofillFeature.canImportFromGooglePasswordManager()) {
+                val config = getConfig()
```
Comment (joshliebe) [inline autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/importing/gpm/feature/AutofillImportPasswordSettings.kt:45]: nit: you could rename this for clarity, I thought it was recursive 😁

**Review by joshliebe — APPROVED** [review]
Tested, LGTM 👍

File: autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/importing/gpm/feature/AutofillImportPasswordSettings.kt (hunk 1 above)
Comment (CDRussell) [reply autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/importing/gpm/feature/AutofillImportPasswordSettings.kt]: Good point; ditched that `with` as it wasn't make the code any clearer and now there is no ambiguity

**Pull request merged**